frontend/src/components/comp-top-table-bar/components/complectation.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import { Draggable } from 'react-beautiful-dnd';
+import { useNavigate } from 'react-router-dom';
+
+import { ComplPreviewInfo, WishlistInput } from '@autoline/shared';
+import { AppRoute } from '@common/enums/enums';
+import { HeartIcon } from '@components/common/icons/icons';
+import { TrashCanIcon } from '@components/common/icons/trash-can/trash-can';
+import { useAppSelector } from '@hooks/hooks';
+import { useDeleteCarFromComparisonMutation } from '@store/queries/comparisons';
+import {
+  useCreateWishlistMutation,
+  useDeleteWishlistMutation,
+} from '@store/queries/preferences/wishlist';
+import { clsx } from 'clsx';
+
+import styles from './styles.module.scss';
+
+export const Complectation = (props: {
+  car: ComplPreviewInfo;
+  index: number;
+}): React.ReactElement => {
+  const authToken = useAppSelector((state) => state.auth.token);
+  const navigate = useNavigate();
+
+  const [isLiked, setIsLiked] = useState(Boolean(props.car.wishlistId));
+  const [isDeleted, setIsDeleted] = useState(false);
+
+  const [createWishlist] = useCreateWishlistMutation();
+  const [deleteWishlist] = useDeleteWishlistMutation();
+  const [deleteCar] = useDeleteCarFromComparisonMutation();
+
+  const handleCreateWishlist = async (data: WishlistInput): Promise<void> => {
+    await createWishlist(data);
+  };
+
+  const handleDeleteWishlist = async (data: WishlistInput): Promise<void> => {
+    await deleteWishlist(data);
+  };
```

Comment:
Please, use context for like functionality. We moved these functions to the single place. Here is `car-list-item` component for the reference
https://github.com/BinaryStudioAcademy/bsa-2022-autoline/pull/198/files#

Reply:
damn thats nice